packages/datagateway-download/src/downloadApiHooks.ts

```diff
@@ -736,3 +738,35 @@ export const useAdminUpdateDownloadStatus = (): UseMutationResult<
     }
   );
 };
+
+/**
+ * Queries the progress of a {@link Download}.
+ * @param prepareId The prepare ID of the {@link Download}.
+ * @param queryOptions Optional `useQuery` option override.
+ */
+export const useDownloadPercentageComplete = <T = DownloadProgress>({
+  preparedId,
+  ...queryOptions
+}: { preparedId: string } & UseQueryOptions<
+  DownloadProgress,
+  AxiosError,
+  T,
+  string[]
+>): UseQueryResult<T, AxiosError> => {
+  const { idsUrl } = React.useContext(DownloadSettingsContext);
+
+  return useQuery(
+    [QueryKey.DOWNLOAD_PROGRESS, preparedId],
+    () =>
+      getPercentageComplete({
+        preparedId: preparedId,
+        settings: { idsUrl },
+      }),
+    {
+      onError: (error) => {
+        handleICATError(error);
```

Comment:
I'll need to check with Chris & Sabah whether they want progress request errors to be broadcast to the user or not (I've just asked Chris just waiting on a response)